src/shared/Microsoft.Git.CredentialManager/Application.cs

```diff
@@ -105,5 +136,114 @@ protected bool WriteException(Exception ex)
 
             return true;
         }
+
+        #region IConfigurableComponent
+
+        string IConfigurableComponent.Name => "Git Credential Manager";
+
+        Task IConfigurableComponent.ConfigureAsync(
+            IEnvironment environment, EnvironmentVariableTarget environmentTarget,
+            IGitConfiguration configuration, GitConfigurationLevel configurationLevel)
+        {
+            // NOTE: We currently only update the PATH in Windows installations and leave putting the GCM executable
+            //       on the PATH on other platform to their installers.
+            if (PlatformUtils.IsWindows())
+            {
+                string directoryPath = Path.GetDirectoryName(_appPath);
+                if (!environment.IsDirectoryOnPath(directoryPath))
+                {
+                    Context.Trace.WriteLine("Adding application to PATH...");
+                    environment.AddDirectoryToPath(directoryPath, environmentTarget);
+                }
+                else
+                {
+                    Context.Trace.WriteLine("Application is already on the PATH.");
+                }
+            }
+
+            string helperKey = $"{Constants.GitConfiguration.Credential.SectionName}.{Constants.GitConfiguration.Credential.Helper}";
+            string gitConfigAppName = GetGitConfigAppName();
+
+            using (IGitConfiguration targetConfig = configuration.GetFilteredConfiguration(configurationLevel))
+            {
+                /*
+                 * We are looking for the following to be considered already set:
+                 *
+                 * [credential]
+                 *     ...                         # any number of helper entries
+                 *     helper =                    # an empty value to reset/clear any previous entries
+                 *     helper = {gitConfigAppName} # the expected executable value in the last position & directly following the empty value
+                 *
+                 */
+
+                string[] currentValues = targetConfig.GetMultivarValue(helperKey, Constants.RegexPatterns.Any).ToArray();
+                if (currentValues.Length < 2 ||
+                    !string.IsNullOrWhiteSpace(currentValues[currentValues.Length - 1]) ||    // second to last entry is empty
```

Comment:
Question on the comments here

Wouldn't the second to last entry be "currentValues.Length -2 ", and the last entry be "currentValues.Length-1"

Reply:
Ah yes! Correct! Will fix and ensure a unit test catches this.

Reply:
Fixed.